web_text_count_field/static/src/js/fields_text_count.js

```diff
@@ -0,0 +1,38 @@
+odoo.define('text_count', function(require) {
+    "use strict";
+
+    var registry = require('web.field_registry');
+    var FieldText = require('web.FieldText');
+
+    var FieldTextCount = FieldText.extend({
+        template: 'FieldTextCount',
+
+        start: function () {
+            this._super();
+            var $textarea = this.$element.find('textarea')[0];
+            $textarea.addEventListener("keyup", this.count_char);
+        },
+
+        count_char: function (e) {
+            var len =  this.value.length;
+            var counter = $(this.parentNode).find('span')[0];
+            counter.innerHTML = len;
+        },
+
+        update_dom: function () {
```

Comment:
do we need this?
also, don't we lock editing when max size is reached?
In that case we should use the `maxlenght` property